### PR TITLE
node.mk: add linting & formatting targets

### PR DESCRIPTION
**Pull request by nathanleiby**

### CODE REVIEW

File: make/node.mk [hunk 1]
```diff
@@ -79,3 +79,35 @@ node-coffee-edit-check:
 node-update-makefile:
 	@wget https://raw.githubusercontent.com/Clever/dev-handbook/master/make/node.mk -O /tmp/node.mk 2>/dev/null
 	@if ! grep -q $(NODE_MK_VERSION) /tmp/node.mk; then cp /tmp/node.mk node.mk && echo "node.mk updated"; else echo "node.mk is up-to-date"; fi
+
+##########################
+## Linting / Formatting ##
+##########################
+# Match .js, .ts, .jsx, and .tsx files. Ignore directories contaning installed dependencies or auto-generated files.
+NODE_ALL_FILES := $(shell find . -regex '.*\.[jt]s[x]\{0,1\}' -not -path "./node_modules/*" -a -not -path "./gen-js/*" -a -not -path "./build/*" -a -not -path "./__build/*")
```
Comment (prime-time) [inline make/node.mk]: what's the `\{0,1\}` part for in the regex?

Comment (prime-time) [inline make/node.mk]: what's the `\{0,1\}` part for in the regex?

Comment (toddobryan-clever) [inline make/node.mk]: Yeah. While I'm pretty sure the regex is right, I'd almost prefer (ts|js|tsx|jsx) for clarity.

**Comment by nathanleiby** [comment]
@toddobryan-clever the problem I ran into is that I couldn't write that kind of regex :/ `find` on Linux doesn't support extended regular expressions (though it does on Mac, via the `-E` flag). hence writing this mess trying to find something that was portable across mac and linux.

I've now updated to something more verbose but simpler to read, I hope 2da12fb

**Pull request merged**